docs: 🏗️ add design of `init-metadata` CLI (#1777)
seedcase-project/seedcase-sprout: docs/design/interface/cli.qmd

instances.
:::

> `seedcase-sprout` has several commands: `extract-metadata`, `compose-metadata`,

Reviewer: How about `create-metadata`?

Author: Hm, I thought of that too, but it isn't completely accurate because it isn't making the metadata, it's making a template that the user would then use to "create" the metadata. :thinking:

Reviewer 2: Hmmmm, `init-metadata`, maybe?

Reviewer 2: However, there's something about this that could sound like initialising or composing the `datapackage.json` file and not a Python script 🤔

Reviewer: I kinda like `init-metadata`. `extract-metadata` also produces a script and not `datapackage.json` 🤷

Author: Agreed! Let's go with `init-metadata` for now, since as @martonvago said it matches the behaviour of `extract-metadata`. We can always change it later.

flowchart LR
    parquet[/"Parquet path<br>[PARQUET-PATH]"/]
    output_opt[/"Output path<br>[--output]"/]
    extract["extract-metadata"]
    output[/"Python script<br>[File]"/]

    parquet --> extract
    output_opt --> extract
    extract --> output
```

### {{< var wip >}} `compose-metadata`

The `compose-metadata` command creates a Python script with metadata fields that
haven't been filled in yet. This command is meant to be used to start filling

> out some metadata when none yet exists. It creates a Python script with empty

Reviewer 2: ```suggestion
haven't been filled in yet. It creates a Python script with empty
```
To me, that sentence doesn't add much, so we could remove it

`*Properties()` dataclasses. It has the signature:

```bash {filename="Terminal"}

> seedcase-sprout compose-metadata [OUTPUT-PATH] --type [TYPE]

Reviewer 2: Does the output path include the file name? Or is that set?
What's the file name when the type is resource?

```

The `--type` determines whether to make a file with metadata for the package
overall (top-level metadata) or for a resource file. The default is the package

> overall.

Reviewer: So if type is `package`, would the script just list a commented-out resource properties (as it does now)?

Author: Yea, exactly :relaxed:

> overall.

Reviewer 2: If type is resource, should it then be (manually?) be added to a package properties script?
